qiskit/transpiler/preset_passmanagers/builtin_plugins.py

```diff
@@ -0,0 +1,288 @@
+
+
+def get_vf2_limit(pass_manager_config):
+    """Get the vf2 call limit for vf2 based layout passes."""
+    vf2_call_limit = None
+    opt_level = pass_manager_config.optimization_level
+    if pass_manager_config.layout_method is None and pass_manager_config.initial_layout is None:
+        if opt_level == 1:
+            vf2_call_limit = int(5e4)  # Set call limit to ~100ms with retworkx 0.10.2
+        elif opt_level == 2:
+            vf2_call_limit = int(5e6)  # Set call limit to ~10 sec with retworkx 0.10.2
+        elif opt_level == 3:
+            vf2_call_limit = int(3e7)  # Set call limit to ~60 sec with retworkx 0.10.2
+    return vf2_call_limit
+
+
+class BasicSwapPassManager(PassManagerStagePlugin):
+    """Plugin class for routing stage with :class:`~.BasicSwap`"""
+
+    def pass_manager(self, pass_manager_config: PassManagerConfig) -> PassManager:
+        """Build routing stage PassManager."""
+        opt_level = pass_manager_config.optimization_level
+        seed_transpiler = pass_manager_config.seed_transpiler
+        target = pass_manager_config.target
+        coupling_map = pass_manager_config.coupling_map
+        backend_properties = pass_manager_config.backend_properties
+        routing_pass = BasicSwap(coupling_map)
+        vf2_call_limit = get_vf2_limit(pass_manager_config)
+        if opt_level == 0:
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map=coupling_map,
+                seed_transpiler=seed_transpiler,
+                use_barrier_before_measurement=True,
+            )
+        elif opt_level == 1:
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map,
+                vf2_call_limit=vf2_call_limit,
+                backend_properties=backend_properties,
+                seed_transpiler=seed_transpiler,
+                check_trivial=True,
+                use_barrier_before_measurement=True,
+            )
+        elif opt_level == 2:
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map=coupling_map,
+                vf2_call_limit=vf2_call_limit,
+                backend_properties=backend_properties,
+                seed_transpiler=seed_transpiler,
+                use_barrier_before_measurement=True,
+            )
+        elif opt_level == 3:
+            return common.generate_routing_passmanager(
+                routing_pass,
+                target,
+                coupling_map=coupling_map,
+                vf2_call_limit=vf2_call_limit,
+                backend_properties=backend_properties,
+                seed_transpiler=seed_transpiler,
+                use_barrier_before_measurement=True,
+            )
+        else:
+            raise TranspilerError(f"Invalid optimization level specified: {opt_level}")
```

Comment:
I would expect this to be captured hire in the call stack, maybe at `pass_manager_config` init time.

Reply:
In the normal case it is, but because `optimization_level` as part of the `PassManagerConfig` was added for this PR I found a few paths where the I had to explicitly add the `optimization_level` to the config object to make this work. I left this in primarily just to guard against other use cases I might have missed or didn't think about.